sds/src/scanner/mod.rs

```diff
@@ -44,30 +45,107 @@ where
     }
 }
 
-pub trait CompiledRuleTrait: Send + Sync {
+pub trait CompiledRuleDyn: Send + Sync {
```

Comment:
Would the previous name still make sense with the current update? (or maybe why `Dyn` keyword better fit now compared to `Trait`)